Neos.ContentRepository/Classes/Domain/Model/AbstractNodeData.php

```diff
@@ -388,6 +388,11 @@ public function getNodeType()
         return $this->nodeTypeManager->getNodeType($this->nodeType);
     }
 
+    public function getRealNodeTypeNameWithoutFallback(): string
```

Comment:
This should be marked internal and how about dropping the `real` and  calling it instead `getNodeTypeNameWithoutFallback` or `getDefinedNodeTypeName` so it doesn't sound so hacky?